core/src/main/java/org/hibernate/ogm/util/impl/TupleContextHelper.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Hibernate OGM, Domain model persistence for NoSQL datastores
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.ogm.util.impl;
+
+import org.hibernate.engine.spi.SessionImplementor;
+import org.hibernate.ogm.dialect.spi.TupleContext;
+import org.hibernate.ogm.model.spi.EntityMetadataInformation;
+import org.hibernate.ogm.persister.impl.OgmEntityPersister;
+
+/**
+ * @author Davide D'Alto
+ */
+public class TupleContextHelper {
+
+	/**
+	 * Given a {@link SessionImplementor} returns the {@link TupleContext} associated to an entity.
+	 *
+	 * @param session the current session
+	 * @param metadata the {@link EntityMetadataInformation} of the entity associated to the TupleContext
+	 * @return the TupleContext associated to the current session for the entity specified, or {@code null} if the
+	 * EntityMetadataInformation is {@code null}
+	 */
+	public static TupleContext tupleContext(SessionImplementor session, EntityMetadataInformation metadata) {
+		if ( metadata != null ) {
```

Comment:
Check `BackendQuery#getSingleEntityMetadataInformationOrNull()`:
```
	/**
	 * Returns the {@link EntityMetadataInformation} of the entity type selected by this query.
	 *
	 * @return the {@link EntityMetadataInformation} of the entity type selected by this query or {@code null} in case this
	 * query does not select exactly one entity type (e.g. in case of scalar values or joins (if supported in future revisions)).
	 */
	public EntityMetadataInformation getSingleEntityMetadataInformationOrNull() {
		return singleEntityMetadataInformation;
	}

```